src/pincode/Pincode.tsx

```diff
@@ -35,20 +35,17 @@ function Pincode({
 
     const newPin = pin + digit
     onChangePin(newPin)
+    if (newPin.length === maxLength) {
+      // Wait for next frame so we the user can see the last digit
+      // displayed before acting on it
+      requestAnimationFrame(() => onCompletePin(newPin))
+    }
   }
 
   const onBackspacePress = () => {
     onChangePin(pin.substring(0, pin.length - 1))
   }
 
-  useEffect(() => {
-    // Wait for next frame so we the user can see the last digit
-    // displayed before acting on it
-    if (pin.length === maxLength) {
-      requestAnimationFrame(() => onCompletePin(pin))
```

Comment:
actually i think this is the problem, apologies i had blindly copied and pasted without checking in my previous comment but here we don't need the `requestAnimationFrame`. we can directly execute `onCompletePin(pin)` since there's no reason to wait. i just tested this locally (without the pincode ref) and it seems to work for this use case, does this also work for you?

Reply:
It looks like the `requestAnimationFrame` is ensuring that the last pin digit is displayed to the user prior to navigating to the next entry; without it sometimes a dot for the 6th digit and the 5th digit is shown.

Reply:
hey sorry for the late reply, this fell off my radar!

i got really curious about this so i dug into this a bit today. i found a few subtle things, i'll elaborate at our 1:1 later today but writing it here also for reference.

the reason why the wrong error is shown for this scenario is because of the way PincodeSet.tsx is handling the verifying and error state. when `onCompletePin2` fails, the verifying state goes back to false so that the the first Pincode screen displays again, because of the `requestAnimationFrame`, the `onCompletePin1` is fired again. that is why we see the "choose a stronger pin" error instead of the "pins don't match" error.

the PincodeSet.tsx file is setting the verifying state via navigation. i have no idea why, but it should be component state. this syncs the error states in `onCompletePin2` by setting the error + verifying state in the same batch. with this change the scenario works correctly for me, but let's try together later!